File: frontend/nutricentro/src/app/core/services/role-service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../enviroment/enviroment';
import {RoleResponseDTO} from '../model/login-model';

@Injectable({
  providedIn: 'root',
})
export class RoleService {
  private http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/v1/roles`;

  // createRole(request: RoleRequestDTO) {
  //   return this.http.post<RoleResponseDTO>(
  //     `${this.apiUrl}/create`,
  //     request
  //   );
  // }

  getRoles() {
    return this.http.get<RoleResponseDTO[]>(
      `${this.apiUrl}/all`
    );
  }
}

File: backend/src/main/java/nutricentro/services/RoleService.java
package nutricentro.services;

import nutricentro.dtos.roles.RoleRequestDTO;
import nutricentro.dtos.roles.RoleResponseDTO;
import nutricentro.entities.RoleEntity;

import java.util.List;
import java.util.Set;

public interface RoleService {
	RoleResponseDTO createRole(RoleRequestDTO request);
	List<RoleResponseDTO> findAll();
	Set<RoleEntity> resolveRoles(Set<String> roles);
}
